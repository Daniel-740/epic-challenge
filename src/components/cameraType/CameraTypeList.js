import { Component } from "react";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import { Button, ButtonGroup } from "@material-ui/core";
import DeleteIcon from "@material-ui/icons/Delete";
import EditIcon from "@material-ui/icons/Edit";
import Box from '@mui/material/Box';

import { connect  } from "react-redux";
import { deleteCameraAction } from "../../store/camera/actions";
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import { Link } from 'react-router-dom';


class List extends Component {

     handleDelete = (id) => {
       this.props.deleteCameraAction(id);
     };
     render() {
          console.log(this.props)
       const { cameraTypes } = this.props;
   
       return (
        <>
         <Grid margin={'1.5rem 1rem'} sx={{display: 'flex', justifyContent: 'end'}}>
            <Grid item xs={4} marginRight="1rem">
               <Link to='/camera-type'>
                    <Button variant="outlined">
                         Add Camera Type
                    </Button>
                </Link>
            </Grid>
          </Grid>      
          <Container maxWidth={false}>
          <Box marginTop={'1rem'}>
          <TableContainer>
            <Table aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell>Id</TableCell>
                  <TableCell>Name</TableCell>
                  <TableCell>Description</TableCell>
                  <TableCell>Actions</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                {cameraTypes.map((row) => (
                  <TableRow key={row.name}>
                    <TableCell component="th" scope="row">
                      {row.id}
                    </TableCell>
                    <TableCell component="th" scope="row">
                      {row.name}
                    </TableCell>
                    <TableCell component="th" scope="row">
                      {row.description}
                    </TableCell>
                    <TableCell component="th" scope="row">
                      <ButtonGroup
                        color="primary"
                        aria-label="outlined primary button group"
                      >
                        <Button variant="contained">
                          <EditIcon />
                        </Button>
                        <Button
                          variant="contained"
                          color="secondary"
                          onClick={() => this.handleDelete(row.id)}
                        >
                          <DeleteIcon />
                        </Button>
                      </ButtonGroup>
                    </TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
          </Box>
          </Container>
        </>
       );
     }
   }
   
   const mapStateToProps = (state) => ({
     cameraTypes: state.camerasReducer.cameraTypes,
   });
   
   const mapDispatchToProps = {
     deleteCameraAction,
   };
   
 export const CameraTypeList = connect(mapStateToProps, mapDispatchToProps)(List);