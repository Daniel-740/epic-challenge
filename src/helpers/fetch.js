
export const findById = (arrayMain, id) => arrayMain.filter( data => data.id === id);

export const checkRelationCameraType = (arrayMain, id) => {
     const response = arrayMain.find(data => data.cameraType === id);

     return response;
}