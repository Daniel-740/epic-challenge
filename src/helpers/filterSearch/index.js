/*
 *
 * @param {string} value: Is string for the value search bar
 * @param {array} arrayFilter: Is array original values for the state redux
 * @param {string} entity: to differentiate the module from the browser, this makes it reusable
 */

export const filterSearch = (value, arrayFilter, entity) => {

     if(entity === 'cameraTypes'){

          const filteredRows = arrayFilter.filter((row) => {
               return row.name.toLowerCase().includes(value.toLowerCase()) || row.description.toLowerCase().includes(value.toLowerCase());
             });

          return filteredRows
     }
/* 
     if(entity === 'cameras'){

     } */

}