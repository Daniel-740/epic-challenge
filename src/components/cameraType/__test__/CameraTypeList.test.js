import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import { CameraTypeList } from '../CameraTypeList';
import {renderWithProviders } from '../../../helpers/test'
import { fireEvent, act } from "@testing-library/react";

const props = {
     cancelSearch: jest.fn(),
     requestSearch: jest.fn(), 
     data: [{
          id: '6465442',
          name: 'dummy-camera',
          description: 'test description'
     }], 
     searched: '',
     navigate: jest.fn(), 
     onDelete: jest.fn()
}

test('render CameraTypeList', () => {
     
     const component = renderWithProviders(<CameraTypeList {...props} />)

     expect(component).toMatchSnapshot();
})

test('click delete Button', () => {
     const component = renderWithProviders(<CameraTypeList {...props} />)
     const button = component.getByRole('button', {name: /delete_button/i});

     fireEvent.click(button);
     expect(props.onDelete).toHaveBeenCalled();
})

test('click edit Button', () => {
     const component = renderWithProviders(<CameraTypeList {...props} />)
     const button = component.getByRole('button', {name: /edit_button/i});

     fireEvent.click(button);
     expect(props.navigate).toHaveBeenCalled();
})

test('search bar value', () => {
     const component = renderWithProviders(<CameraTypeList {...props} />)
     const searchBar = component.getByPlaceholderText('Search...');

     fireEvent.change(searchBar, { target: { value: 'test' } })
  
     expect(props.requestSearch).toHaveBeenCalled()
     expect(searchBar.value).toBe('test')
})