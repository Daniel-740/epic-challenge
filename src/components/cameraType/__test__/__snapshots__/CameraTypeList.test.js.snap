// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CameraTypeList 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root css-10ydfi0-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-17xat2w-MuiGrid-root"
        >
          <a
            href="/camera-type"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Add Camera Type
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
      <div
        class="MuiContainer-root css-ur2jdm-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-15g2oxy"
        >
          <div
            class="MuiPaper-root ForwardRef-root-1 mb2 mt2 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="ForwardRef-searchContainer-7"
            >
              <div
                class="MuiInputBase-root MuiInput-root ForwardRef-input-6 MuiInputBase-fullWidth MuiInput-fullWidth"
              >
                <input
                  class="MuiInputBase-input MuiInput-input"
                  placeholder="Search..."
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-searchIconButton-4"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root ForwardRef-icon-5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-iconButtonHidden-3"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root ForwardRef-icon-5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiTableContainer-root"
          >
            <table
              aria-label="simple table"
              class="MuiTable-root"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Id
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Name
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Description
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Actions
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-body"
                    role="cell"
                    scope="row"
                  >
                    6465442
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-body"
                    role="cell"
                    scope="row"
                  >
                    dummy-camera
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-body"
                    role="cell"
                    scope="row"
                  >
                    test description
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-body"
                    role="cell"
                    scope="row"
                  >
                    <div
                      aria-label="outlined primary button group"
                      class="MuiButtonGroup-root"
                      role="group"
                    >
                      <button
                        aria-label="edit_button"
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-containedPrimary"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-label="delete_button"
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-containedSecondary"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </th>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root css-10ydfi0-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-17xat2w-MuiGrid-root"
      >
        <a
          href="/camera-type"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add Camera Type
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
    <div
      class="MuiContainer-root css-ur2jdm-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-15g2oxy"
      >
        <div
          class="MuiPaper-root ForwardRef-root-1 mb2 mt2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="ForwardRef-searchContainer-7"
          >
            <div
              class="MuiInputBase-root MuiInput-root ForwardRef-input-6 MuiInputBase-fullWidth MuiInput-fullWidth"
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                placeholder="Search..."
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-searchIconButton-4"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root ForwardRef-icon-5"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-iconButtonHidden-3"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root ForwardRef-icon-5"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiTableContainer-root"
        >
          <table
            aria-label="simple table"
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Id
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Description
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  6465442
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  dummy-camera
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  test description
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  <div
                    aria-label="outlined primary button group"
                    class="MuiButtonGroup-root"
                    role="group"
                  >
                    <button
                      aria-label="edit_button"
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-containedPrimary"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-label="delete_button"
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-containedSecondary"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </th>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
